lib/charms/loki_k8s/v0/loki_push_api.py

```diff
@@ -1885,7 +1889,7 @@ def _on_relation_departed(self, _: RelationEvent) -> None:
             self._container.stop(WORKLOAD_SERVICE_NAME)
         self.on.log_proxy_endpoint_departed.emit()
 
-    def _get_container(self, container_name: str = "") -> Container:
+    def _get_container(self, container_name: str = "") -> Container:  # pyright: ignore
```

Comment:
Just curious why the ignore here?

Reply:
Because the function technically returns an `Optional[Container]`; however, changing the return type means adding an `if` EVERY time we call the `self._container` property, which is a lot; for the purpose of keeping the PR small, I just ignored it here

If we want to refactor this specifically, I'd suggest a new issue, because it's a bit of code which will need to be tested better